### PR TITLE
7kaa: new recipe

### PR DESCRIPTION
**Pull request by TheZeldakatze**
Is the package name fine? Because whilst it has this name in the gpo and in the debian package repository I think that it is not descriptive of the game. I hope the icon is fine

It should compile on x86_64 and x86_gcc4.

### CODE REVIEW

**Comment by Begasus** [comment]
> Is the package name fine?

Name is fine, one way to check is with Repology, also the path you should use (games/stratagy) is fine, we mostly compare with Gentoo, you can see that here: https://repology.org/project/7kaa/versions

File: games-strategy/7kaa/7kaa-2.15.4p1.recipe [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+CHECKSUM_SHA256="f0265c077025d77b435f1c10b8ba178bdb16329e51ab735330ebaa71a104fbf8"
+PATCHES="7kaa-2.15.4p1.patchset"
+ADDITIONAL_FILES="7kaa.rdef.in"
+
+ARCHITECTURES="all !x86_gcc2"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	7kaa$secondaryArchSuffix
+	app:sevenKingdoms
```
Comment (Begasus) [inline games-strategy/7kaa/7kaa-2.15.4p1.recipe:22]: Maybe?
```diff
- 	app:sevenKingdoms
+ 	app:SevenKingdoms
```

File: games-strategy/7kaa/7kaa-2.15.4p1.recipe [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+	lib:libopenal$secondaryArchSuffix
+	lib:libSDL2_2.0$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	cmd:curl_config$secondaryArchSuffix
+	devel:libopenal$secondaryArchSuffix
+	devel:libcurl$secondaryArchSuffix
+	devel:libsdl2$secondaryArchSuffix
```
Comment (Begasus) [inline games-strategy/7kaa/7kaa-2.15.4p1.recipe:37]: This should be the same as with the lib:*
```diff
- 	devel:libsdl2$secondaryArchSuffix
+ 	devel:libSDL2_2.0$secondaryArchSuffix
```

File: games-strategy/7kaa/7kaa-2.15.4p1.recipe [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+	cmd:g++$secondaryArchSuffix
+	cmd:make
+	"
+
+BUILD() {
+	# for some reason the configure script doesn't find these variables by itself
+	CURL_CFLAGS=$(curl-config --cflags)
+	CURL_LIBS=$(curl-config --libs)
+	export CURL_CFLAGS
+	export CURL_LIBS
```
Comment (Begasus) [inline games-strategy/7kaa/7kaa-2.15.4p1.recipe:53]: From github I see there is a configure.ac file there, maybe run `autoreconf -vfi runConfigure ...` ?

Comment (TheZeldakatze) [reply games-strategy/7kaa/7kaa-2.15.4p1.recipe:53]: The official release source has no autoconf files, though I could theoretically use the github repository

**Comment by TheZeldakatze** [comment]
I reused the description from the debian packet repo, is doing something like that fine? If not I could try to write a description myself though it probably would be terrible

File: games-strategy/7kaa/7kaa-2.15.4p1.recipe (hunk 3 above)
Comment (Begasus) [reply games-strategy/7kaa/7kaa-2.15.4p1.recipe:53]: OK, haven't checked a build yet, will report back later :)

**Comment by Begasus** [comment]
> I reused the description from the debian packet repo, is doing something like that fine? If not I could try to write a description myself though it probably would be terrible

You could see what is used in the readme or from their github page (haven't checked there) so for now it should be ok :)

File: games-strategy/7kaa/7kaa-2.15.4p1.recipe [hunk 4]
```diff
@@ -0,0 +1,86 @@
+SUMMARY="Seven Kingdoms: Ancient Adversaries"
+DESCRIPTION="Seven Kingdoms, designed by Trevor Chan, brings a unique blend of Real-Time Strategy\
+	with the addition of trade, diplomacy, and espionage.\
+	\
+	The game enables players to compete against up to six other kingdoms allowing players to conquer \
+	opponents by defeating them in war (with troops or machines), capturing their buildings with spies, \
+	or offering opponents money for their kingdom."
```
Comment (Begasus) [inline games-strategy/7kaa/7kaa-2.15.4p1.recipe:7]: DESCRIPTION can be at most 100 characters now, a few lines exceed this number.
No tabs in the beginning of  new line. Space at the end of a line before the backslash.
If you want to split the paragraphs, no backslash at the end or in the new line: https://github.com/haikuports/haikuports/wiki/HaikuPorter-Guidelines#description

**Comment by Begasus** [comment]
Had a quick look at their github page for disabling fullscreen (lucky it doesn't change the screen resolution when quit), but can't find a quick fix for this, it's not too bad, but if you could find a fix it would be nice (don't like having no control over using/not using fullscreen) ;)

File: games-strategy/7kaa/7kaa-2.15.4p1.recipe [hunk 5]
```diff
@@ -0,0 +1,86 @@
+SUMMARY="Seven Kingdoms: Ancient Adversaries"
+DESCRIPTION="Seven Kingdoms, designed by Trevor Chan, brings a unique blend of Real-Time Strategy\
+	with the addition of trade, diplomacy, and espionage.\
+	\
```
Comment (Begasus) [inline games-strategy/7kaa/7kaa-2.15.4p1.recipe:4]: No tabs here (in front of the lines) and in the next, new line can be added by skipping backslashes, plus space at the end of the first line (check after a build how HaikuDepot shows it:
```diff
-DESCRIPTION="Seven Kingdoms, designed by Trevor Chan, brings a unique blend of Real-Time Strategy\
-	with the addition of trade, diplomacy, and espionage.\
-	\
+DESCRIPTION="Seven Kingdoms, designed by Trevor Chan, brings a unique blend of Real-Time Strategy \
+with the addition of trade, diplomacy, and espionage.
+
```

File: games-strategy/7kaa/7kaa-2.15.4p1.recipe [hunk 6]
```diff
@@ -0,0 +1,86 @@
+SUMMARY="Seven Kingdoms: Ancient Adversaries"
+DESCRIPTION="Seven Kingdoms, designed by Trevor Chan, brings a unique blend of Real-Time Strategy\
+with the addition of trade, diplomacy, and espionage.\
+\
```
Comment (Begasus) [inline games-strategy/7kaa/7kaa-2.15.4p1.recipe:4]: Line 2 space before the backslash, line 3 and 4, no backslashes (looks better to start line 5 on a new line), you could even leave out line 4.
```diff
-DESCRIPTION="Seven Kingdoms, designed by Trevor Chan, brings a unique blend of Real-Time Strategy\
-with the addition of trade, diplomacy, and espionage.\
-\
+DESCRIPTION="Seven Kingdoms, designed by Trevor Chan, brings a unique blend of Real-Time Strategy \
+with the addition of trade, diplomacy, and espionage.
```

**Comment by Begasus** [comment]
Thanks for taking the time to make the changes :+1:

**Comment by TheZeldakatze** [comment]
> Thanks for taking the time to make the changes 👍

And thanks for taking the time to review everything

**Pull request merged**